dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/datavalueset/DefaultDataValueSetService.java

```diff
@@ -1053,11 +1056,16 @@ private ImportContext createDataValueSetImportContext(
       DataValueSet data,
       BatchHandler<DataValue> dataValueBatchHandler,
       BatchHandler<DataValueAudit> auditBatchHandler) {
-    final User currentUser = currentUserService.getCurrentUser();
+
+    String currentUsername = CurrentUserUtil.getCurrentUsername();
+    User currentUser = userService.getUserByUsername(currentUsername);
```

Comment:
this pattern seems to come up a lot:
- get current username
- then get current user (using current username from above)

could they be their own method to reduce code?

Reply:
Yes, this is something I want to refine further, I was trying to find the most common pattern as I was refactoring here. I want to do this in a follow-up PR. Also, getting rid of more of the user == null means super user.